ssl/ssl_ciph.c

```diff
@@ -1622,25 +1900,90 @@ STACK_OF(SSL_CIPHER) *ssl_create_cipher_list(const SSL_METHOD *ssl_method,
                 OPENSSL_free(co_list);
                 sk_SSL_CIPHER_free(cipherstack);
                 OSSL_TRACE_CANCEL(TLS_CIPHER);
+                OPENSSL_free(tmp_flags);
                 return NULL;
             }
             if (trc_out != NULL)
                 BIO_printf(trc_out, "<%s>\n", curr->cipher->name);
+            tmp_flags[i] = curr->flags;
+            if (curr->group_id)
+                tmp_flags[i] |= CIPHER_IN_GROUP;
+            if (group_id_pref && group_id_pref != curr->group_id)
+                /* Beginning of new group, remove flag from previous cipher. */
+                tmp_flags[i - 1] &= ~CIPHER_IN_GROUP;
+            i++;
+            group_id_pref = curr->group_id;
         }
     }
+    /* Remove flag from last cipher in cipher_list. */
+    tmp_flags[i - 1] &= ~CIPHER_IN_GROUP;
+
     OPENSSL_free(co_list);      /* Not needed any longer */
     OSSL_TRACE_END(TLS_CIPHER);
 
-    if (!update_cipher_list_by_id(cipher_list_by_id, cipherstack)) {
+    tmp_cipher_list_flags = ssl_cipher_list_flags_new(NULL);
+    if (tmp_cipher_list_flags == NULL) {
         sk_SSL_CIPHER_free(cipherstack);
+        OPENSSL_free(tmp_flags);
+        return NULL;
+    }
+    tmp_cipher_list_flags->cipher_list = cipherstack;
+    ciphers_cnt = sk_SSL_CIPHER_num(cipherstack) * sizeof(uint8_t);
+    tmp_cipher_list_flags->flags = OPENSSL_malloc(ciphers_cnt);
+    if (tmp_cipher_list_flags->flags == NULL) {
+        ssl_cipher_list_flags_free(tmp_cipher_list_flags);
+        OPENSSL_free(tmp_flags);
+        return NULL;
+    }
+    memcpy(tmp_cipher_list_flags->flags, tmp_flags, ciphers_cnt);
+    OPENSSL_free(tmp_flags);
+    tmp_flags = NULL;
+    if (*cipher_list_flags != NULL) {
+        ssl_cipher_list_flags_free(*cipher_list_flags);
+    }
+    *cipher_list_flags = tmp_cipher_list_flags;
+
+    if (!update_cipher_list_by_id(cipher_list_by_id, cipherstack)) {
+        ssl_cipher_list_flags_free(*cipher_list_flags);
+        *cipher_list_flags = NULL;
         return NULL;
     }
-    sk_SSL_CIPHER_free(*cipher_list);
-    *cipher_list = cipherstack;
 
     return cipherstack;
 }
 
+char *SSL_CIPHER_flags_description(const SSL *s, int i, char * buf, int len)
+{
+    SSL_CIPHER_FLAGS *cipher_list_flags = SSL_get_ciphers_flags(s);
+    uint8_t *flags;
+    char *group = " ";
+    char prefer = ' ';
+    static const char *format = "%s %c ";
+
+    if (buf == NULL) {
+        len = 8;
+        if ((buf = OPENSSL_malloc(len)) == NULL) {
+            SSLerr(SSL_F_SSL_CIPHER_DESCRIPTION, ERR_R_MALLOC_FAILURE);
```

Comment:
This function code is detected as a mismatch. However, we do not use function codes any more (upcoming in 3.0), so the simplest thing you can do is have a `0` instead.

Reply:
Fixed, replaced with `0`